### ISSUE TITLE
Can't level up pokémons

### ISSUE THREAD

**Issue by terrenzio** [NECROBOTIO/NecroBot#4447]
Hi guys. I want to level up some pokémons with the bot with these settings:
"AutomaticallyLevelUpPokemon": true,
  "OnlyUpgradeFavorites": true,
  "UseLevelUpList": false,
  "AmountOfTimesToUpgradeLoop": 5,
  "GetMinStarDustForLevelUp": 5000,
  "LevelUpByCPorIv": "iv",
  "UpgradePokemonCpMinimum": 1000.0,
  "UpgradePokemonIvMinimum": 95.0,
  "UpgradePokemonMinimumStatsOperator": "and",
Even if there are pokémon that have this settings, the bot won't do anything. Can someone help me?

**Comment by Linus9000** [comment]
Look into your log files, maybe it has upgraded but didn't tell you in the console.

see also #4436

**Comment by terrenzio** [comment]
Nope.
There's no "upgrade" in log.

**Comment by DBa2016** [comment]
> "OnlyUpgradeFavorites": true

Are the pokemons you want to upgrade favorited?

**Comment by terrenzio** [comment]
Yes, they are.

**Comment by DBa2016** [comment]
Are they also over iv 95 and cp 1000?

You would see "Pokemon Upgraded:..." in the console when a pokemon is upgraded.

**Comment by terrenzio** [comment]
Yes, they are and no, I can't see that in console

**Comment by terrenzio** [comment]
I have also set  
"OnlyUpgradeFavorites": true,
  "UseLevelUpList": false,
  "AmountOfTimesToUpgradeLoop": 1,
  "GetMinStarDustForLevelUp": 50,
  "LevelUpByCPorIv": "iv",
  "UpgradePokemonCpMinimum": 10.0,
  "UpgradePokemonIvMinimum": 9.0,
  "UpgradePokemonMinimumStatsOperator": "and",
But nothing change

**Comment by Nikolovo** [comment]
same problem here, even with my backup config from 0.8.2 (auto levelup worked there perfectly)

**Comment by nfettinger** [comment]
Have you already gotten any help using Discord Chat? This allows us to rule out configuration issues BEFORE wasting our time looking at the code.

**Comment by terrenzio** [comment]
Asked kinda 100 times.. Someone tried help me but without any results. I heard other people who has the same problem

**Comment by spamfighter82** [comment]
yeah, i also tried that for about an hour before i gave up and did it in nox

**Comment by KurtKoko** [comment]
I don´t know what your problem is, but mine got upgraded. Just like i wanted.

I faved all the Pkmn i wanted to get an upgrad, changed the config.json and voila! Everything like a charme. Nothing to see in the console, but i can see it in the logfiles.
The console only shows the Pkmn wich got "EVOLVED"


 
```

   "AutomaticallyLevelUpPokemon": true,
   "OnlyUpgradeFavorites": true,
   "UseLevelUpList": true,
   "AmountOfTimesToUpgradeLoop": 5,
   "GetMinStarDustForLevelUp": 5000,
   "LevelUpByCPorIv": "iv",
   "UpgradePokemonCpMinimum": 900.0,
   "UpgradePokemonIvMinimum": 90.0,
   "UpgradePokemonMinimumStatsOperator": "and",
```

```
  "PokemonsToLevelUp": [
    "chansey",
    "gengar",
    "omanyte",
    "vaporeon",
    "jolteon",
    "flareon",
    "snorlax",
    "scyther",
    "magmar",
    "electabuzz",
    "gyarados",
    "lapras",
    "dragonite"
],
```

**Comment by terrenzio** [comment]
Ok I will try but it's not a good solution. In this case it will level up all the pokémon in the list and you have to manually write all the pokémon you want level up. That's a bot's work. By the way, thanks for the moment :+1:

**Comment by KurtKoko** [comment]
It works for me even with 
"UseLevelUpList": false,

I just made the list, because i don't want all my fav pkmn to be leveled up ^^

**Comment by terrenzio** [comment]
For me doesn't work. i will retry now

**Comment by KurtKoko** [comment]
I am sry. I double checked it now again and for me it does not work either. I have to use
"UseLevelUpList": true, 
Otherwise it wouldn't work. Really annoying.

**Comment by terrenzio** [comment]
Also tried with the list. No way

**Comment by KurtKoko** [comment]
At least the list should work.

**Comment by KurtKoko** [comment]
Cross-check with my config. Maybe there is something in your config that doesn´t fit right ?

ATTENTION:
Don´t use this as your config before you didn´t double-checked it thoroughly! 
[config.txt](https://github.com/NECROBOTIO/NecroBot/files/411370/config.txt)

**Comment by terrenzio** [comment]
I compared my config with yours. Nothing special between the two config. I dunno why won't work.

**Comment by userix** [comment]
Having the issue as well.  I tried using the list function as well, and still the one pokemon I have that fits the criteria still isn't getting updated.

**Comment by Mikey9191** [comment]
i am also having the same problem, not a noob here and i know what i am doing, so far for me the upgrade feature is broken since i upgraded to 0.8.5

**Comment by fuqua123** [comment]
yup having the same problem here, upgrading wont work, nothing in console indicating upgrading and nothing in the log files.

here is my config:

"AutomaticallyLevelUpPokemon": true,
  "OnlyUpgradeFavorites": true,
  "UseLevelUpList": false,
  "AmountOfTimesToUpgradeLoop": 5,
  "GetMinStarDustForLevelUp": 5000,
  "LevelUpByCPorIv": "iv",
  "UpgradePokemonCpMinimum": 100.0,
  "UpgradePokemonIvMinimum": 100.0,
  "UpgradePokemonMinimumStatsOperator": "and",

my intension is for it to only upgrade IV100 pokemons that i have favorited.

**Comment by Linus9000** [comment]
I can now confirm that when UseLevelUpList is false it won't upgrade any pokemon. Why that is i cannot tell, but the logic in the PoGo.NecroBot.Logic/Tasks/LevelUpPokemonTask.cs looks right. Maybe the task isn't executed in that case? I don't know where that usually happens so I can't look any furter..

**Comment by noodle-** [comment]
This seems to be fixed by https://github.com/NECROBOTIO/NecroBot/commit/31faadba2af006562d61a9d5dd37128d708cb5a0

AmountOfTimesToUpgradeLoop is still ignored though.

**Comment by fuqua123** [comment]
i have tried to set "UseLevelUpList": false, to true but no difference.

**Comment by vison99** [comment]
It worked for me with  

"OnlyUpgradeFavorites": false,
"UseLevelUpList": false.

It doesn't show in the command line. But when I checked they are leveled.

**Comment by terrenzio** [comment]
@vison99 can you please send your config onliy for level up?
